opensearch/src/http/aws_auth.rs

```diff
@@ -0,0 +1,82 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+use std::time::SystemTime;
+
+use aws_sigv4::{
+    http_request::{
+        sign, PayloadChecksumKind, PercentEncodingMode, SignableBody, SignableRequest,
+        SignatureLocation, SigningParams, SigningSettings,
+    },
+    signing_params::BuildError,
+};
+use aws_types::{
+    credentials::{ProvideCredentials, SharedCredentialsProvider},
+    region::Region,
+    Credentials,
+};
+use reqwest::Request;
+
+const SERVICE_NAME: &str = "es";
+
+fn get_signing_params<'a>(
+    credentials: &'a Credentials,
+    region: &'a Region,
+) -> Result<SigningParams<'a>, BuildError> {
+    let mut settings = SigningSettings::default();
```

Comment:
could this be 

```
let settings = SigningSettings {
 percent_encoding_mode : PercentEncodingMode::Double;
 payload_checksum_kind : PayloadChecksumKind::NoHeader;
 signature_location : SignatureLocation::Headers;
 expires_in : None, 
 ..Default::default()
};

Reply:
`SigningSettings` is non_exhaustive, so unfortunately can not be constructed with a struct expression.

Reply:
Though on closer inspection the values I'm setting are the defaults, so will simplify it.